js/components/file.js

```diff
@@ -52,16 +52,76 @@ Fliplet.FormBuilder.field('file', {
   created: function() {
     Fliplet.FormBuilder.on('reset', this.onReset);
   },
+  updated: function() {
+    if (this.readonly) {
+      var $vm = this;
+      var isFileDataLoaded = false;
+      var filesID = _.map(this.value, function(fileURL) {
+        if (typeof fileURL === 'string') {
+          return +fileURL.match(/v1\/media\/files\/([0-9]+)/)[1];
+        }
+
+        isFileDataLoaded = true;
+
+        return null;
+      });
+
+      if (isFileDataLoaded) {
+        return;
+      }
+
+      Fliplet.Media.Files.getAll({
+        files: filesID,
+        fields: ['name', 'url', 'metadata', 'createdAt']
+      }).then(function(files) {
+        $vm.value = files;
+      }).catch(function(err) {
+        console.log(Fliplet.parseError(err));
```

Comment:
@YaroslavOvdii You told me that if `Fliplet.Media.Files.getAll()` is called with multiple file IDs and one or more of the files is not found, the promise is rejected. I asked you to confirm whether the error contains information on which of the files are missing. What was the answer?

@squallstar If it doesn't contain the information on which files (of the provided file IDs) are missing, can that information be provided? We would need that to continue the operation and remove the missing files from the list of files to be displayed (instead of triggering an error alert).

Reply:
@tonytlwu I've looked into this issue more closely and that what I've discovered:
When we pass not existing ID or `null, NaN, undefined, 0`
Then response will return only existing values.
In that light, I suggest simply remove console.log here, because it wouldn't give any information to the user.